src/LLVMSupport/SMTConstraintGenerator.cpp

```diff
@@ -317,7 +414,8 @@ partitionInstructionsInBlock(vector<Instruction *> instructions) {
     Internals::constrainInstruction(this, i);
   }
 
-  auto mod = Internals::incremementalPartitioning(this, instructions);
+  // auto mod = Internals::incremementalPartitioning(this, instructions);
+  auto mod = Internals::binaryPartitioning(this, instructions);
```

Comment:
TODO for me: parametrize the choice of search strategy as part of the SMT config